pgrx/src/fcall.rs

```diff
@@ -0,0 +1,143 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use crate::{
+    direct_function_call, list::PgList, pg_sys, pg_sys::AsPgCStr, Array, FromDatum, IntoDatum,
+};
+
+pub trait FCallArg {
```

Comment:
Incorrect implementations of this could lead to UB, even for something as simple as just lying about the OID, thus it must be

```suggestion
pub unsafe trait FCallArg {
```

Reply:
If IntoDatum didn’t consume then we wouldn’t need this trait at all.